src/algebra/monoid_algebra.lean

```diff
@@ -103,6 +103,34 @@ instance : semiring (monoid_algebra k G) :=
     sum_add],
   .. finsupp.add_comm_monoid }
 
+variables {R : Type*} [semiring R]
+
+/-- A non-commutative version of `monoid_algebra.lift`: given a additive homomorphism `f : k →+ R`
+and a multiplicative monoid homomorphism `g : G →* R`, returns the additive homomorphism from
+`monoid_algebra k G` such that `lift_nc f g (single a b) = f b * g a`. If `f` is a ring homomorphism
+and the range of either `f` or `g` is in center of `R`, then the result is a ring homomorphism.  If
+`R` is a `k`-algebra and `f = algebra_map k R`, then the result is an algebra homomorphism called
+`monoid_algebra.lift`. -/
+def lift_nc (f : k →+ R) (g : G →* R) : monoid_algebra k G →+ R :=
+lift_add_hom (λ x : G, (add_monoid_hom.mul_right (g x)).comp f)
+
+@[simp] lemma lift_nc_single (f : k →+ R) (g : G →* R) (a : G) (b : k) :
+  lift_nc f g (single a b) = f b * g a :=
+lift_add_hom_apply_single _ _ _
+
+@[simp] lemma lift_nc_one (f : k →+* R) (g : G →* R) : lift_nc (f : k →+ R) g 1 = 1 :=
+by simp [one_def]
+
+lemma lift_nc_mul_noncomm (f : k →+* R) (g : G →* R)
```

Comment:
How about just calling this `lift_nc_mul`?

Reply:
Done.